com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/SetBreakpointsRequestHandler.java

```diff
@@ -36,15 +40,21 @@
 import com.microsoft.java.debug.core.protocol.Types;
 
 public class SetBreakpointsRequestHandler implements IDebugRequestHandler {
+
+    private static final Logger logger = Logger.getLogger(Configuration.LOGGER_NAME);
+
     private BreakpointManager manager = new BreakpointManager();
 
+    private IDebugAdapterContext context = null;
+
     @Override
     public List<Command> getTargetCommands() {
         return Arrays.asList(Command.SETBREAKPOINTS);
     }
 
     @Override
     public CompletableFuture<Response> handle(Command command, Arguments arguments, Response response, IDebugAdapterContext context) {
+        this.context = context;
```

Comment:
Instead of saving the context as a member, suggest making it an argument of `reinstallBreakpoints` in a more functional way.

Reply:
For functional's parameter, the parameter capture the local variable in its body. But as for the context, it is a global variable and doesn't change in this functional calling lifecycle.